src/doc/style-guide/src/expressions.md

```diff
@@ -831,20 +831,56 @@ let arr = [combinable(
     an_expr,
     another_expr,
 )];
+
+let x = Thing(an_expr, another_expr, match cond {
+    A => 1,
+    B => 2,
+});
+
+let x = format!("Stuff: {}", [
+    an_expr,
+    another_expr,
+]);
 ```
 
 Apply this behavior recursively.
 
-For a function with multiple arguments, if the last argument is a multi-line
-closure with an explicit block, there are no other closure arguments, and all
-the arguments and the first line of the closure fit on the first line, use the
-same combining behavior:
+If the last argument is a multi-line closure with an explicit block,
+only apply the combining behavior if there are no other closure arguments.
```

Comment:
Do we want to keep this property? Here's an example that this wouldn't allow, but that seems *relatively* reasonable:

```rust
func(arg, || expr, |closure_arg| {
    long_closure_here();
    more_of_the_long_closure();
});
```

We might want to prohibit this formatting, but let's consider this case specifically before assuming we want to preserve this restriction.